rules/CodeQuality/Rector/ClassMethod/ResponseReturnTypeControllerActionRector.php

```diff
@@ -172,6 +186,48 @@ private function refactorResponse(ClassMethod $classMethod): Node
             return $classMethod;
         }
 
-        return $classMethod;
+        return $this->refatorWithNew($classMethod);
+    }
+
+    private function refatorWithNew(ClassMethod $classMethod): ?ClassMethod
+    {
+        // early check
+        if (! $this->betterNodeFinder->hasInstancesOf($classMethod, [New_::class])) {
+            return null;
+        }
+
+        $returns = $this->betterNodeFinder->findReturnsScoped($classMethod);
+        if (! $this->returnAnalyzer->hasOnlyReturnWithExpr($classMethod, $returns)) {
+            return null;
+        }
+
+        $returnedTypes = [];
+        foreach ($returns as $return) {
+            if (! $return->expr instanceof Node\Expr) {
+                // already validated above
+                throw new ShouldNotHappenException();
+            }
+
+            $returnedTypes[] = $this->getType($return->expr);
```

Comment:
better to use native type on this case, ensuring no typo by docblock definition:

```suggestion
            $returnedTypes[] = $this->nodeTypeResolver->getNativeType($return->expr);
```

Reply:
This is also one of cases, where we are more tolerant as return type comes from framework itself.

* Controller, with `#[Route()]` or `@Route()`, with response return type.